Build failure: u8string_view → string_view conversion error in alstring.h with C++23

## Build failure: `u8string_view` → `string_view` conversion error in `alstring.h` with C++23

### Description

Build fails with C++23 due to a type mismatch in `3rdparty/openal/common/alstring.h`.

The function `char_as_u8` declares a return type of `std::string_view` but returns `std::u8string_view`. This implicit conversion was allowed in C++20 but is a hard error in C++23, where `char8_t` is a distinct type.

### Error

```
C:\...\openal\common\alstring.h(48): error C2440: 'return': cannot convert from 'std::u8string_view' to 'std::basic_string_view<char,std::char_traits<char>>'
note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
```

### Root Cause

In `3rdparty/openal/common/alstring.h`, the function has a mismatched return type:

```cpp
// Current (broken in C++23):
inline auto char_as_u8(const std::string_view str LIFETIMEBOUND) -> std::string_view
{ return std::u8string_view{reinterpret_cast<const char8_t*>(str.data()), str.size()}; }
```

### Fix

Change the return type to match the returned value:

```cpp
inline auto char_as_u8(const std::string_view str LIFETIMEBOUND) -> std::u8string_view
{ return std::u8string_view{reinterpret_cast<const char8_t*>(str.data()), str.size()}; }
```

This fix has been tested and confirmed working.

### Environment

- OS: Windows 11
- Compiler: MSVC 19.x
- IDE: CLion 2026.1.1
- `WINDOWS_SDK_VERSION`: 10.0.26100.0

--- a/3rdparty/openal/common/alstring.h
+++ b/3rdparty/openal/common/alstring.h
@@ -44,7 +44,7 @@ inline auto char_as_u8(const std::string_view str LIFETIMEBOUND) -> std::string_
 inline auto u8_as_char(const std::string_view str LIFETIMEBOUND) -> std::string_view
 { return str; }
 #else
-inline auto char_as_u8(const std::string_view str LIFETIMEBOUND) -> std::string_view
+inline auto char_as_u8(const std::string_view str LIFETIMEBOUND) -> std::u8string_view
 { return std::u8string_view{reinterpret_cast<const char8_t*>(str.data()), str.size()}; }
 
 inline auto u8_as_char(const std::u8string_view str LIFETIMEBOUND) -> std::string_view

